### PR TITLE
added explanations for "learning_rate", "weight_decay" and "n_epochs"

### LINKED ISSUE #5: Machine learning parameter explanations
**As a** user, **I want** have clear explanations of machine learning parameters when I rest my mouse on them, **so that** I can understand their effects on model training.

### Acceptance Criteria

- [x] Add hover-over explanations for the learning rate, weight decay, and number of epochs parameters;
- [x] Ensure that the explanations are written in simple, non-technical language understandable to users who are not familiar with machine learning.;

### Additional Resources:

image

### Draft explanations:

Learning Rate:

Explanation: The learning rate determines how quickly or slowly the model learns from data. A higher learning rate may lead to faster learning but could cause the model to overshoot the optimal solution. Conversely, a lower learning rate may result in slower learning but is safer and more likely to find the best solution.

Weight decay:

Explanation: Weight decay helps prevent overfitting by penalizing large parameter values in the model. Increasing weight decay encourages the model to learn simpler patterns from the data, improving its ability to generalize to new, unseen examples.

Number of Epochs:

Explanation: The number of times the entire dataset is passed forward and backward through the machine learning model during training. Increasing the number of epochs allows the model to see the data more times, potentially improving its accuracy. However, too many epochs can lead to overfitting, where the model memorizes the training data instead of learning generalizable patterns.

### Notes

Effort: 3

### PR DESCRIPTION
**Pull request by robertkrasniqi**
Resolves #5
I have saved the texts from the issue https://github.com/claassenlab/cellpose/issues/5#issue-2269173363 as an array and added code so that they are displayed to the user when the user hovers over the respective text.

### CODE REVIEW

File: cellpose/gui/guiparts.py [hunk 1]
```diff
@@ -223,10 +223,15 @@ def __init__(self, parent, model_strings):
         # choose parameters
         labels = ["learning_rate", "weight_decay", "n_epochs", "model_name"]
         self.edits = []
+        self.parameter_explanations = ["The learning rate determines how quickly or slowly the model learns from data. A higher learning rate may lead to faster learning but could cause the model to overshoot the optimal solution. Conversely, a lower learning rate may result in slower learning but is safer and more likely to find the best solution.",
+                     "Weight decay helps prevent overfitting by penalizing large parameter values in the model. Increasing weight decay encourages the model to learn simpler patterns from the data, improving its ability to generalize to new, unseen examples.",
```
Comment (JoshuaSchlucke) [inline cellpose/gui/guiparts.py]:
```suggestion
                     "Weight decay helps prevent overfitting by penalizing large parameter values in the model. \n Increasing weight decay encourages the model to learn simpler patterns from the data, improving its ability to generalize to new, unseen examples.",

```

**Comment by JoshuaSchlucke** [comment]
Sieht super aus, hast es besser gemacht als ich

**Review by ShalinGhe — APPROVED** [review]
Super Lösung gefunden ohne viel am alten Code zu ändern! 
Lässt sich schnell verstehen und funktioniert einwandfrei, good job!

**Review by sonjajk — APPROVED** [review]
Ich kann mich meinen Vorgängern nur anschließen. Sehr gute Lösung!

**Review by JoshuaSchlucke — CHANGES_REQUESTED** [review]
würde es für bessere "readability" nur bisschen brechen

**Review by JoshuaSchlucke — APPROVED** [review]
